ext/har/logger.go

```diff
@@ -0,0 +1,110 @@
+package har
+
+
+import (
+    "encoding/json"
+    "net/http"
+    "os"
+    "sync"
+    "time"
+
+    "github.com/elazarl/goproxy"
+)
+
+// Logger implements a HAR logging extension for goproxy
+type Logger struct {
+	mu             sync.Mutex
+	har            *Har
+	captureContent bool
+}
+
+// NewLogger creates a new HAR logger instance
+func NewLogger() *Logger {
+	return &Logger{
+		har: New(),
+	}
+}
+
+// OnRequest handles incoming HTTP requests
+func (l *Logger) OnRequest(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+	// Store the start time in context for later use
+	if ctx != nil {
+		ctx.UserData = time.Now()
+	}
+	return req, nil
+}
+
+// OnResponse handles HTTP responses
+func (l *Logger) OnResponse(resp *http.Response, ctx *goproxy.ProxyCtx) *http.Response {
+	if resp == nil || ctx == nil || ctx.Req == nil || ctx.UserData == nil {
+		return resp
+	}
+
+	startTime, ok := ctx.UserData.(time.Time)
+	if !ok {
+		return resp
+	}
+
+	// Create HAR entry
+	entry := Entry{
+		StartedDateTime: startTime,
+		Time:           time.Since(startTime).Milliseconds(),
+		Request:        ParseRequest(ctx.Req, l.captureContent),
+		Response:       ParseResponse(resp, l.captureContent),
+		Cache:          Cache{},
+		Timings: Timings{
+			Send:    0,
+			Wait:    time.Since(startTime).Milliseconds(),
+			Receive: 0,
+		},
+	}
+
+	// Add server IP
+	entry.FillIPAddress(ctx.Req)
+
+	// Add to HAR log thread-safely
+	l.mu.Lock()
+	l.har.AppendEntry(entry)
+	l.mu.Unlock()
+
+	return resp
+}
+
+// SetCaptureContent enables or disables request/response body capture
+func (l *Logger) SetCaptureContent(capture bool) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	l.captureContent = capture
+}
+
+// SaveToFile writes the current HAR log to a file
+func (l *Logger) SaveToFile(filename string) error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	encoder := json.NewEncoder(file)
+	encoder.SetIndent("", "  ")
+	return encoder.Encode(l.har)
+}
+
+// Clear resets the HAR log
+func (l *Logger) Clear() {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	l.har = New()
+}
+
+// GetEntries returns a copy of the current HAR entries
+func (l *Logger) GetEntries() []Entry {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	entries := make([]Entry, len(l.har.Log.Entries))
+	copy(entries, l.har.Log.Entries)
+	return entries
+}
```

Comment:
The HAR exporting part must be redesigned from scratch, here there are a bunch of issues.
Let me know if you want to do that, or in case I can fix this personally.

Basically, instead of this, the most common use will be the periodically send (e.g. every 5 minutes) of the HAR entries to an external logger API service or graphana or whatever.
So we have to accept a function from the user that, periodically, will consume the list of entries, pass it to this user function, and delete it from our queue.
The user will do whatever he wants with this list of entries, because he passed us its own function.

A goroutine will read from the entries channel and append to a local slice.
For the function call, there can be two options that the user can select:
- every N requests we call the function (we implement this using a counter)
- every time.Duration (e.g. 5 minutes) we call the user function

Let me know about this

Reply:
I'm actually fixing all the issues you have pointed out right now,  I can do this part tonight, but if not tomorrow. I haven't worked with go routines before, but I think I understand the general idea for the saving function. thank you.

Reply:
Yeah don't worry, it's holiday time, so feel free to take your time.
Your help is really appreciated here!
It's not an urgent feature.

Reply:
nah I should be thanking you! the feedback is appreciated. 

> Yeah don't worry, it's holiday time, so feel free to take your time. Your help is really appreciated here! It's not an urgent feature.